GitUI/UserControls/RevisionGrid/RevisionGridControl.RevisionGridInMemFilter.cs

```diff
@@ -66,8 +66,8 @@ static bool CheckCondition(string? filter, Regex? regex, string? value)
             {
                 if (string.IsNullOrEmpty(authorFilter) &&
                     string.IsNullOrEmpty(committerFilter) &&
-                    string.IsNullOrEmpty(messageFilter) &&
-                    !ObjectId.IsValidPartial(messageFilter, minLength: 5))
+                    (GitExtensions.Strings.IsNullOrEmpty(messageFilter) ||
+                    !ObjectId.IsValidPartial(messageFilter, minLength: 5)))
```

Comment:
Please review this. The previous logic seems broken. If `messageFilter` is null or empty, it would never be a valid partial. I'm pretty sure this was supposed to be `||`.

Reply:
change seem OK

Reply:
Can we please have this change in a separate commit for ease of tracking/bisecting, if that becomes necessary.

Reply:
0981ff16676be8461c3e8e80050630857ff6f0d1